aten/src/ATen/native/Distributions.cpp

```diff
@@ -295,4 +295,30 @@ Tensor _s_dirichlet_cpu(const Tensor& alpha, Generator *gen) {
   });
   return ret;
 }
+
+Tensor multinomial_cpu(const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
+  Tensor result = at::empty({0}, self.options().dtype(kLong));
+  multinomial_out_cpu(result, self, n_sample, with_replacement, gen);
+  return result;
+}
+
+Tensor& multinomial_out_cpu(Tensor& result, const Tensor& self, int64_t n_sample, bool with_replacement, Generator *gen) {
+  TORCH_CHECK(self.dim() >= 1, "multinomial only supports at least 1D tensor, got: ", self.dim(), "D");
+  TORCH_CHECK(at::isFloatingType(self.scalar_type()), "multinomial only supports floating-point dtypes for input, got: ", self.scalar_type());
+  TORCH_CHECK(result.scalar_type() == ScalarType::Long, "multinomial expects Long tensor out, got: ", result.scalar_type());
+  TORCH_CHECK(n_sample > 0, "cannot sample n_sample <= 0 samples");
+  int64_t n_categories = self.size(-1);
+  TORCH_CHECK(with_replacement || (n_sample <= n_categories), "cannot sample n_sample > prob_dist.size(-1) samples without replacement");
+  if (self.dim() > 1) {
```

Comment:
so if `self.dim() > 2`....we just ignore the other dimensions?

Reply:
I think this PR is for migrating multinomial from TH to ATen. Currently we have different behavior of this function on CPU and CUDA for dims > 2. On CUDA we throw an exception: "prob_dist must be 1 or 2 dim". On CPU we return incorrect tensor (https://github.com/pytorch/pytorch/issues/12309). We need to discuss what will be the right solution for this, but I think it should be in separate PR.
For now, I just return tensor with "correct" values even if dim>2

Reply:
Negative indexes also prevent problems with squeeze/unsqueeze